### PR TITLE
Add support for Visual Studio csc c# compiler

### PR DESCRIPTION
**Pull request by thiblahute**
(based on https://github.com/mesonbuild/meson/pull/2248)

### CODE REVIEW

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -991,10 +994,16 @@ def generate_cs_target(self, target, outfile):
...
+            deps.append(os.path.normpath(rel_src))
 
-        elem = NinjaBuildElement(self.all_outputs, outputs, 'cs_COMPILER', rel_srcs)
+        for dep in target.get_external_deps():
+            commands.extend_direct(dep.get_link_args())
+        commands += self.build.get_project_args(compiler, target.subproject)
+        commands += self.build.get_global_args(compiler)
+
+        elem = NinjaBuildElement(self.all_outputs, outputs, 'cs_COMPILER', rel_srcs,
+                allow_backslashes=True)
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E128] continuation line under-indented for visual indent

**Comment by jpakkane** [comment]
#2248 was merged so please update to remove the duplicate tommits.

**Comment by thiblahute** [comment]
> #2248 was merged so please update to remove the duplicate tommits.

Done.

**Comment by jpakkane** [comment]
Tests for this should be enabled in CI. A quick check showed that at least the resource compiler test fails. It would be cool to get VS support as well, but I'd say just having the Ninja backend is enough for now.

**Comment by thiblahute** [comment]
I tested the tests and got them working here, we check again, and how should I get them running on the CI server?

**Comment by jpakkane** [comment]
AFAICT the C# compiler should be available in the CI machines. You'll need to change `run_project_tests.py` to run the tests if MS C# compiler is in path. Possibly only if the backend is ninja. You'll probably want to do a standalone function for that instead of hacking it inside the current line.

You'll also need to change the dependency check to use `required: false` and then print the skipping magic string if it is not found.

**Comment by thiblahute** [comment]
Let's see how it goes :-)

**Comment by jpakkane** [comment]
Not well, apparently. :(

**Comment by jpakkane** [comment]
What's the status of this? It would be good to get this in.

**Comment by jpakkane** [comment]
ping

**Comment by jpakkane** [comment]
This is now part of #3132 so closing.

**Pull request closed without merging**